### DISCUSSION TITLE
Add option to pull images in Stacks and Deployments

### DISCUSSION THREAD

**Discussion by FibreTTP** [moghtech/komodo#99] [Ideas]
After requesting an option to _disable_ automatic image pulling, I now come with a feature request to _add_ an option to manually pull images in Stack configuration and Deployment pages (and perhaps the image pages themselves).

The button should appear next to the Deploy, Stop, Destroy, etc., buttons, and be labeled either "Pull" or "Update". For stacks, this should simply call `docker compose pull`.

This would allow me to pull images for running services not pinned to any version, which I do not know have an update or not, and choose to Redeploy after if needed, as to not Redeploy and cause unnecessary downtime if there wasn't even an update.

Since the `docker compose pull` command allows specifying a service, a dialog box or dropdown could also be implemented, allowing users to choose which service's image should be pulled, with an option to pull all of them.

Consider using the flag `--ignore-pull-failures`.

**Comment by mbecker20** [comment]
Its a good idea, I can definitely implement a `PullDeployment` / `PullStack` execution. Additionally, it gives me idea, I think Deployment / Stacks could use this to notify for image updates.

1.  Add `poll_image` configuration, if the true then will call `PullDeployment` / `PullStack` on an interval, say every 5 minutes
2. Keep track of image hashes.

The UI can then display to the user that there is an update available, and the new image will already be pulled onto target server and ready for deploy.

3. With additional `auto_update` configuration, Core can automatically trigger redeploy when there is an update available.

**Reply by kingp0dd** [reply]
May i ask what the options in the current version are for:

```
Auto Update
Poll for Updates
Check for updates to the image on an interval.

ENABLED
auto update
Trigger a redeploy if a newer image is found.


```

Is this not for image updates?

**Reply by FibreTTP** [reply]
Yes, they are. Are you asking for more detail on what they do?

**Reply by mbecker20** [reply]
Inspired by this thread, I implemented this feature already starting from 1.16.9, those options you see are the implementation of this feature.

**Reply by kingp0dd** [reply]
That was fast, might have missed it in the docs. Thank you!
